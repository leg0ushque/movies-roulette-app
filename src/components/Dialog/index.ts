import Dialog, { type IDialogProps } from './Dialog';

export type { IDialogProps };

export default Dialog;
