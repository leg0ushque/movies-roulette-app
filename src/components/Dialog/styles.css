:root {
    --modal-common-padding: 5%;
}

.dialog-overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5);
    z-index: 100;
}

.dialog:not(.dialog.notification) {
    top: 50%;
}

.dialog {
    z-index: 101;
    position: fixed;
    left: 50%;
    
    width: 81.3%;
    padding: 0 0 3% 0;

    background-color: var(--color3);
    font-family: 'Montserrat';
    color: var(--color5);

    transform: translate(-50%, -50%);
}

.dialog.notification {
    top: 30%;
    width: 50.17%;
    text-align: center;
}

.dialog-header {
    padding: 10px;
    margin-bottom: 15px;
}

.dialog-logo {
    padding: 0 0 0 0;
    margin: 0 0 15px 0;

    width: 50px;
    border-radius: 100%;
    border: 0;

    text-align: center;
    background-color: var(--color1);
    font-family: 'Montserrat';
    color: var(--color5);
    line-height: 50px;

    font-size: xx-large;
}

.dialog-logo {
    display: block;
    margin-left: auto;
    margin-right: auto;
}

.dialog-title {
    font-size: xx-large;
    text-transform: uppercase;
    margin-bottom: 2vh;
}

.dialog:not(.notification) .dialog-title {
    margin-left: var(--modal-common-padding);
}

.dialog-close-button {
    float: right;
    border: 0;
    background-color: var(--color3);
    color: var(--color5);
    font-size: x-large;
}

.dialog:not(.notification) .dialog-body {
    padding: 20px var(--modal-common-padding);
}

/*=============*/

.dialog:not(.dialog.notification) .dialog-body {
    max-height: 60vh;
    overflow-y: scroll;
}

/*=============*/

.dialog-body {
    padding: 20px 3%;
}

.dialog-body:last-child {
    padding-bottom: 0;
}