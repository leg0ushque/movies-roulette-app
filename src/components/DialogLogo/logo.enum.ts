enum Logo {
  Plus = '+',
  Check = 'âœ“',
  Question = '?'
}

export default Logo
