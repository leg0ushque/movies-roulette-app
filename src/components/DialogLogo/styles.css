.dialog-logo {
  padding: 0 0 0 0;
  margin: 0 0 15px 0;

  width: 50px;
  border-radius: 100%;
  border: 0;

  text-align: center;
  background-color: var(--color1);
  font-family: 'Montserrat';
  color: var(--color5);
  line-height: 50px;

  font-size: xx-large;
  display: block;
  margin-left: auto;
  margin-right: auto;
}
