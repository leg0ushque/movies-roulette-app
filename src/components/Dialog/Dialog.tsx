import './styles.css';

import React from 'react';

export interface IDialogProps {
  logo: string
  title: JSX.Element | string
  isNotification: boolean
  onClose: () => void
}

const logo = (content: string | null): JSX.Element | null => {
  if (content !== null) {
    return (
      <div className="dialog-logo">
        {content}
      </div>
    )
  } else {
    return null
  }
}

const Dialog: React.FC<React.PropsWithChildren<IDialogProps>> = (props) => {
  return (
    <div>
        <div className="dialog-overlay" />
        <div className={'dialog' + (props.isNotification ? ' notification ' : '')}>
          <div className="dialog-header">
              <button className="dialog-close-button" onClick={props.onClose}>&#10005;</button>
          </div>
          {logo(props.logo)}
          <div className="dialog-title">
            {props.title}
            </div>
          <div className="dialog-body">
            {props.children}
          </div>
        </div>
    </div>
  );
}

export default Dialog;
