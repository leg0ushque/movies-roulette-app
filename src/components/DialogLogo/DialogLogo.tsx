import './styles.css';

import React from 'react';

import type Logo from './logo.enum';

interface IDialogLogoProps {
  char?: Logo
}

const DialogLogo: React.FunctionComponent<IDialogLogoProps> = ({ char }) => {
  if (!char) {
    return null
  }

  return (
    <div className="dialog-logo prevent-select" role="dialog-logo">
      { char }
    </div>
  )
}

export default DialogLogo;
