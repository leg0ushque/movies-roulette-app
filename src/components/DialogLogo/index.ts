import DialogLogo from './DialogLogo';
import Logo from './logo.enum';

export { Logo };

export default DialogLogo
