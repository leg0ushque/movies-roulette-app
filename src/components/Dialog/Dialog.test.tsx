import '@testing-library/jest-dom/extend-expect';

import React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';

import { Logo } from '../DialogLogo';
import Dialog from './Dialog';

describe('Dialog', () => {
  const onCloseMock = jest.fn();

  const TEST_TITLE: JSX.Element = (<h2>test title</h2>)
  const TEST_TITLE_TEXT = '<h2>test title</h2>'
  const TEST_BODY: JSX.Element = (<div><span>test body</span></div>)
  const TEST_BODY_TEXT = '<div><span>test body</span></div>'

  test('has all items passed in props rendered', () => {
    render(
      <Dialog
        logo={Logo.Check}
        title={TEST_TITLE}
        isWide={true}
        isCentered={false}
        onClose={onCloseMock}
      >
        {TEST_BODY}
      </Dialog>)

    const dialogTitle = screen.getByRole('dialog-title');
    const dialogBody = screen.getByRole('dialog-body')

    expect(dialogTitle).toMatchSnapshot(TEST_TITLE_TEXT);
    expect(dialogBody).toMatchSnapshot(TEST_BODY_TEXT);
  });

  test('has to be centered when proper value passed in props', () => {
    render(
      <Dialog
        title={TEST_TITLE}
        isCentered={true}
        onClose={onCloseMock}
      >
        {TEST_BODY}
      </Dialog>)

    const dialog = screen.getByRole('dialog')

    expect(dialog).toHaveClass('centered');
  });

  test('has to be wider when proper value passed in props', () => {
    render(
      <Dialog
        title={TEST_TITLE}
        isWide={true}
        isCentered={true}
        onClose={onCloseMock}
      >
        {TEST_BODY}
      </Dialog>)

    const dialog = screen.getByRole('dialog')

    expect(dialog).toHaveClass('wide');
  });

  test('has scrollable body when proper value passed in props', () => {
    render(
      <Dialog
        title={TEST_TITLE}
        hasScrollableBody
        onClose={onCloseMock}
      >
        {TEST_BODY}
      </Dialog>)

    const dialogBody = screen.getByRole('dialog-body')

    expect(dialogBody).toHaveClass('scrollable');
  });

  test('has onClose callback called on close button pressed', () => {
    const { getByRole } = render(
      <Dialog
        title={TEST_TITLE}
        isWide={true}
        isCentered={false}
        onClose={onCloseMock}
      >
        {TEST_BODY}
      </Dialog>)

    const closeButton = getByRole('dialog-close-button')

    fireEvent.click(closeButton);

    expect(onCloseMock).toBeCalledTimes(1);
  });

  test('has no logo rendered when it\'s not passed in props', () => {
    const { container } = render(
      <Dialog
        title={TEST_TITLE}
        isWide={true}
        isCentered={false}
        onClose={onCloseMock}
      >
        {TEST_BODY}
      </Dialog>)

    const dialogLogo = container.querySelector('.dialog-logo')

    expect(dialogLogo).toBeNull();
  });
});
