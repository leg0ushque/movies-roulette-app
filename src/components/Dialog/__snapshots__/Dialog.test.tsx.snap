// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog has all items passed in props rendered: <div><span>test body</span></div> 1`] = `
<div
  class="dialog-body"
  role="dialog-body"
>
  <div>
    <span>
      test body
    </span>
  </div>
</div>
`;

exports[`Dialog has all items passed in props rendered: <h2>test title</h2> 1`] = `
<div
  class="dialog-title"
  role="dialog-title"
>
  <h2>
    test title
  </h2>
</div>
`;
